docs/technical_reference/unit_models/boron_removal.rst

```diff
@@ -0,0 +1,164 @@
+                  'proton_name': 'name_of_species_representing_protons',  #[is optional]
+                  'hydroxide_name': 'name_of_species_representing_hydroxides', #[is optional]
+
+                  'caustic_additive':
+                  {
+                    'additive_name': 'name_of_the_actual_chemical', #[is optional]
+                    'cation_name': 'name_of_cation_species_in_additive', #[is optional]
+                    'mw_additive': (value, units), #[is required]
+                    'charge_additive': value, #[is required]
+                  }
+               }
+
+For example, this 'chem_dict' would be passed into the model on construction as
+one of the configuration options as shown below.
+
+.. code-block::
+
+    model.fs.unit = BoronRemoval(
+            default={
+                "property_package": model.fs.properties,
+                "chemical_mapping_data": chem_dict,
+            }
+        )
+
+**NOTE: The above example assumes you have already constructed a pyomo model named 'model' and attached an IDAES flowsheet named 'fs' to it, as well as a properties block named 'properties'**
+
+Equations and Relationships
+---------------------------
+Reaction constants :math:`K_{w}` and :math:`K_{a}` are taken from the Benjamin (2010) chemistry
+textbook (see References below).
+
+.. csv-table::
+   :header: "Description", "Equation"
+
+   "Electroneutrality Constraint", ":math:`C_{H} = C_{OH} + C_{A} + Alk_{Res} - n_{caustic}*C_{caustic}`"
+   "Residual Alkalinity Expression", ":math:`Alk_{Res} = -{\sum_{i} n_i \cdotp C_{out,i} }`"
+   "Total Boron Balance Constraint", ":math:`C_{in,HA} + C_{in,A} = C_{HA} + C_{A}`"
+   "Water Dissociation Constraint", ":math:`K_{w} = C_{H} \cdotp C_{OH}`"
+   "Boron Dissociation Constraint", ":math:`K_{a} \cdotp C_{HA} = C_{H} \cdotp C_{A}`"
+   "Component (for Boron, Borate, Protons, and Hydroxide) mass balance", ":math:`0 = - M_{i,out} + C_{i} \cdotp Q`"
+   "Caustic cation mass balance", ":math:`0 = M_{caustic,in} - M_{caustic,out} + S_{caustic}`"
+
+   "Caustic cation gain rate", ":math:`S_{caustic} = Q \cdotp \frac{D}{MW}`"
+
+**NOTE:** :math:`Q` **is defined as the total volumetric flow rate and** :math:`S_{j}` **is the source/sink term for component** :math:`j`
+
+**NOTE:** :math:`C_{H}` **is the proton concentration,** :math:`C_{OH}` **is the hydroxide concentration,** :math:`C_{A}` **is the Borate concentration, and** :math:`C_{HA}` **is the Boron concentration.**
+
+**NOTE:** :math:`n_{i}` **is ion charge of component** :math:`i`
+
+**NOTE:** :math:`C_{in,i}` **is the concentration from the inlet stream and** :math:`C_{out,i}` **is the resulting concentration in the outlet stream**
+
+**NOTE:** :math:`Alk_{Res}` **is the approximate total alkalinity of the stream after the first RO stage and is summed over all ions that are not hydroxide or borate ions (which are already included in the electroneutrality balance)**

```

Comment:
I still don't think this is clear enough.  If the model is ran on a solution containing "conservative ions (those with little buffering ability / unaffected by pH)", the equation would not be right to calculate the alkalinity.  This should probably be clarified. Also is the "stream after the first RO" be the inlet stream of this unit? If so , would that be right for the charge balance equation?

Reply:
(1) You bring up a good point. This calculation is not the true alkalinity, because these species would change as pH changes. However, we are simplifying this analysis to focus solely on boron dissociation. Presumably, most all other ions that contribute to the buffering capacity would have been removed from the 1st RO stage, so any changes in their concentrations would be minimal compared to changes in boron/borate speciation. 

Maybe it would be best to rename or reword this. This is mostly what ever are the residual conservative ions in solution (the ones that are not explicitly being calculated for). This is related to alkalinity (that is why I labeled it as 'residual alkalinity'), but is not the actual solution alkalinity. 

(2) Because the species (specifically anything not H+, OH-, Boron, and Borate) in this calculation are assumed unchanged from inlet to outlet, their concentrations inside the domain are the same as the concentrations from the 1st RO stage. 

Maybe it would be best to label this term as 'Remaining Charge' rather than residual Alkalinity.